pkg/controller/envoy/ads_event.go

```diff
@@ -146,10 +146,11 @@ func (svc *ServiceEvent) handleCdsResponse(rsp *service_discovery_v3.DiscoveryRe
 			svc.DynamicLoader.ClusterCache.SetCdsHash(cluster.GetName(), newHash)
 			log.Debugf("[CreateApiClusterByCds] update cluster %s, status %d, cluster.type %v",
 				cluster.GetName(), apiStatus, cluster.GetType())
-			svc.DynamicLoader.CreateApiClusterByCds(apiStatus, cluster)
 		} else {
 			log.Debugf("unchanged cluster %s", cluster.GetName())
+			apiStatus = core_v2.ApiStatus_UNCHANGED
 		}
+		svc.DynamicLoader.CreateApiClusterByCds(apiStatus, cluster)
```

Comment:
It will do some unnecessary work if this is unchanged. As we discussed, we can move the `clusterNames` setup out of `CreateApiClusterByCds` to here

Reply:
> It will do some unnecessary work if this is unchanged. As we discussed, we can move the `clusterNames` setup out of `CreateApiClusterByCds` to here

Yes, it's better performance; However, it seems that CreateApiListenerByLds is not suitable for moving out Routenames. To keep  the logic of CreateApiClusterByCds is consistent with that of CreateApiListenerByLds, clusterNames is not moved out. The replacement solution is to check the value of apiStatus in CreateApiClusterByCds，If the value is ApiStatus_UNCHANGED, the cache is not updated. Whether the current solution is appropriate？